### PR TITLE
[WIP] Configure custom domains

### PR DESCRIPTION
**Pull request by tarteo**
image
image

### CODE REVIEW

File: argocd_website/static/src/js/portal.js [hunk 1]
```diff
@@ -42,4 +42,74 @@ odoo.define("argocd_website.portal", function (require) {
...
+            this.checkDomainCNAME();
+        },
+
+        _onButtonClick: function () {
+            this.checkDomainCNAME();
+        },
+
+        _onKeyDown: function () {
+            this.$input.removeClass("is-invalid");
+            this.$input.removeClass("is-valid");
```
Comment (ByteMeAsap) [inline argocd_website/static/src/js/portal.js:71]: you can include both the classes at once separated by space removeClass("is-invalid is-valid")

File: argocd_sale/tests/test_invoice_to_application.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        partner = self.env["res.partner"].create({"name": "Onestein B.V. (some here)"})
+        product = self.env.ref(
+            "argocd_sale.demo_curq_basis_product_template"
+        ).product_variant_ids
+        orders = self.env["sale.order"].create(
+            {
+                "partner_id": partner.id,
+                "order_line": [
+                    Command.create(
+                        {"name": "Product", "product_id": product.id, "price_unit": 0}
```
Comment (ByteMeAsap) [inline argocd_sale/tests/test_invoice_to_application.py]: Add [0] on product at product.id just to be sure and for readers

File: argocd_website/static/src/js/portal.js [hunk 3]
```diff
@@ -42,4 +42,74 @@ odoo.define("argocd_website.portal", function (require) {
...
+        },
+
+        checkDomainCNAME: function () {
+            var self = this;
+            var domain = this.$input.val();
+
+            if (!domain) {
+                self.$input.removeClass("is-invalid");
+                self.$input.removeClass("is-valid");
```
Comment (ByteMeAsap) [inline argocd_website/static/src/js/portal.js:80]: you can include both the classes at once separated by space removeClass("is-invalid is-valid")

**Comment by tarteo** [comment]
TODO: Forgot to check access rights for portal users

**Comment by tarteo** [comment]
I want to add A record support

**Review by MrGigSolutions — APPROVED** [review]
Approved for testing.

**Pull request merged**